### PR TITLE
[Attach][Macro] Fix issues identified with an attached macro

### PR DESCRIPTION
**Pull request by Tishj**
https://github.com/duckdblabs/duckdb-internal/issues/3368

1. `query_table` was fully quoting the entire path, because of this it couldn't support qualified paths. We now use `QualifiedName::Parse` to extract the path and then stringify this appropriately (quoting where necessary)
2. `Bind(TableFunctionRef &ref)` did not bind the schema/catalog, so it was mistakenly treating `checksum_macro` (from the test) as the schema instead of the catalog.

### CODE REVIEW

File: test/sql/catalog/function/attached_macro.test [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# name: test/sql/catalog/function/attached_macro.test
+# group: [function]
+
+require skip_reload
+
+require httpfs
+
+statement ok
+ATTACH 'https://blobs.duckdb.org/data/checksum-macro.duckdb' AS checksum_macro;
```
Comment (carlopi) [inline test/sql/catalog/function/attached_macro.test:9]: I think this works the same (= was broken in the same way) also for local dbs, so I would probably remove the httpfs dependency

**Review by carlopi — COMMENTED** [review]
Left a comment in that httpfs and remote db can be dropped, since otherwise it will eventually break

Great for this to be fixed and tested.

**Review by Mytherin — APPROVED** [review]
Thanks! LGTM

**Comment by Tishj** [comment]
@Mytherin can we rerun <https://github.com/duckdb/duckdb/actions/runs/11706653687/job/32604243142?pr=14715> ?
I don't think this is actually caused by this PR

**Comment by Mytherin** [comment]
I think we can just merge this, thanks!

**Pull request merged**